src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,227 @@
+        taskList.incrementTaskNum();
+        displayTasksNum();
+        try{
+            storage.updateFile(taskList.getStoredTasks());
+        } catch (IOException e) {
+            System.out.println("Oops! Something broke: " + e);
+        }
+    }
+    private static void addTodoTask(String[] commandLine) {
+
+        try {
+            validateTodo(commandLine);
+        } catch (IllegalDukeArgumentException e) {
+            System.out.println("Oh no! The description of a todo cannot be empty.");
+            return;
+        }
+        Todo td = new Todo(commandLine[1]);
+        taskList.addItem(td);
+        addTaskMessage();
+        System.out.println("  " + td.getTypeIcon() + td.getStatusIcon() + " " + td.getDescription());
+        taskList.incrementTaskNum();
+        displayTasksNum();
+        try {
+            storage.updateFile(taskList.getStoredTasks());
+        } catch (IOException e) {
+            System.out.println("Oops! Something broke: " + e);
+        }
+    }
+
+    private static void validateTodo(String[] commandLine) throws IllegalDukeArgumentException{
+        if (commandLine.length == 1) {
+            throw new IllegalDukeArgumentException();
+        }
+    }
+
+    private static void addDeadlineTask(String[] commandLine) {
+        String[] deadlineString = commandLine[1].split("/by");
+        Deadline dl = new Deadline(deadlineString[0], deadlineString[1]);
+        taskList.addItem(dl);
+        addTaskMessage();
+        System.out.println("  " + dl.getTypeIcon() +
+                dl.getStatusIcon() + " " + dl.getDescription() + "(by: " + dl.getBy() + ")");
+        taskList.incrementTaskNum();
+        displayTasksNum();
+        try{
+            storage.updateFile(taskList.getStoredTasks());
+        } catch (IOException e) {
+            System.out.println("Oops! Something broke: " + e);
+        }
+    }
+    private static void findItem(String[] commandLine) {
+        String keyword = commandLine[1];
+        int count = 1;
+        System.out.println("Here are the matching tasks in your list: ");
+        for (Task i : taskList.getStoredTasks()) {
+            if (i.getDescription().contains(keyword)) {
+                if (i instanceof Deadline) {
+                    System.out.println(count + "." + i.getTypeIcon() + i.getStatusIcon() + i.getDescription() + "/by: " + i.getInfo());
+                } else if (i instanceof Event) {
+                    System.out.println(count + "." + i.getTypeIcon() + i.getStatusIcon() + i.getDescription() + "/start, end: " + i.getInfo());
+                } else {
+                    System.out.println(count + "." + i.getTypeIcon() + i.getStatusIcon() + i.getDescription() + i.getInfo());
+                }
+                count++;
+            }
+        }
+    }
+    private static void addTaskMessage() {
+        System.out.println("Got it. I've added this task.");
+    }
+    private static void listMessage() {
```

Comment:
Methods like this could be structured within a seperate message handling class. Try to only keep relevant methods within their respective classes.